### ISSUE TITLE
Git SCM fails when :local_user contains a space

### ISSUE THREAD

**Issue by fnordfish** [capistrano/capistrano#1738]
#### Steps to reproduce

1. Use the GIT SCM strategy
2. Setup the `:local_user` variable so that it contains a full name (eg: `Robert Schulze`. actualy, I was reading it from `ENV["GIT_COMMITTER_NAME"] || ENV["GIT_AUTHOR_NAME"]`)
3. Try to deploy

#### Expected behaviour

The `git-ssh` wrapper script should be callable.

#### Actual behaviour

```
 DEBUG Uploading /tmp/git-ssh-project-staging-Robert Schulze.sh 0.0%
  INFO Uploading /tmp/git-ssh-project-staging-Robert Schulze.sh 100.0%
  INFO [8bce1bd1] Running /usr/bin/env chmod 700 /tmp/git-ssh-project-staging-Robert Schulze.sh as user@host
...
DEBUG [8bce1bd1] 	chmod:
 DEBUG [8bce1bd1] 	cannot access ‘/tmp/git-ssh-project-staging-Robert’
 DEBUG [8bce1bd1] 	: No such file or directory
 DEBUG [8bce1bd1]
 DEBUG [8bce1bd1] 	chmod:
 DEBUG [8bce1bd1] 	cannot access ‘Schulze.sh’
 DEBUG [8bce1bd1] 	: No such file or directory
 DEBUG [8bce1bd1]
```

Looking at the target host, I can see, that the file is correctly created: `/tmp/git-ssh-project-staging-Robert Schulze.sh`.

**Comment by mattbrictson** [comment]
Thanks for the detailed report! I agree, this is a bug. Would you be comfortable contributing a PR? I think a `gsub` call to replace non-alphanumerics with hyphens when constructing the `:git_wrapper_path` would be a reasonable solution.

https://github.com/capistrano/capistrano/blob/b751b68b8d99c05236f744be5da49e42ddff8c06/lib/capistrano/tasks/git.rake#L9-L10

**Comment by fnordfish** [comment]
Sorry, but I actually won't be able to make the time soon. So, this one is up for grabs :)

**Comment by mattbrictson** [comment]
OK, first come first served, whoever wants to take it! 😄

**Comment by will-in-wi** [comment]
The fix itself is trivial, but I'm not sure where to write a test for it. Any suggestions?

Also, is this still relevant after https://github.com/capistrano/capistrano/pull/1572 goes in?

**Comment by mattbrictson** [comment]
> is this still relevant after #1572 goes in?

Yes. I'll need to make sure it is not clobbered by the merge of #1572.

**Comment by mattbrictson** [comment]
Fixed via #1742
